crates/matrix-sdk-indexeddb/src/event_cache_store/migrations.rs

```diff
@@ -0,0 +1,46 @@
+use super::{indexeddb_serializer::IndexeddbSerializer, keys, Result};
+use indexed_db_futures::{
+    idb_object_store::IdbObjectStoreParameters,
+    request::{IdbOpenDbRequestLike, OpenDbRequest},
+    IdbDatabase, IdbKeyPath, IdbVersionChangeEvent,
+};
+use wasm_bindgen::JsValue;
+
+const CURRENT_DB_VERSION: u32 = 1;
+
+pub async fn open_and_upgrade_db(
+    name: &str,
+    _serializer: &IndexeddbSerializer,
+) -> Result<IdbDatabase> {
+    let mut db = IdbDatabase::open(name)?.await?;
+
+    let old_version = db.version() as u32;
+
+    if old_version == 0 {
+        // TODO some temporary code just to get going
```

Comment:
Big TODO, not sure where things will be going, this is mostly just a placeholder to get the ball rolling.